### PR TITLE
Documentation mobility model, cahn-hilliard filtering

### PR DESCRIPTION
**Pull request by PierreLaurentinCS**
# Description of the problem

- Documentation for the mobility models and phase filtering for Cahn-Hilliard equations

### CODE REVIEW

File: include/core/mobility_cahn_hilliard_model.h [hunk 1]
```diff
@@ -20,57 +20,66 @@
 #include <core/interface_property_model.h>
 
 /**
- * @brief Abstract class that allows to calculate the
- * mobility for the Cahn-Hilliard-Navier-Stokes equations.
+ * @brief Implementation of a general case computation of the mobility
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:23]: What do you mean by general case here? It's a bit hard to understand.

File: include/core/mobility_cahn_hilliard_model.h [hunk 2]
```diff
@@ -20,57 +20,66 @@
...
+ * mobility constant.
  */
 class MobilityCahnHilliardModelConstant : public MobilityCahnHilliardModel
 {
 public:
   /**
    * @brief Default constructor
+   *
+   * @param[in] p_mobility_cahn_hilliard_constant the user chosen mobility
+   * constant.
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:66]:
```suggestion
   * @param[in] p_mobility_cahn_hilliard_constant the user defined mobility
   * constant.
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 3]
```diff
@@ -90,9 +100,10 @@ class MobilityCahnHilliardModelConstant : public MobilityCahnHilliardModel
   }
 
   /**
-   * @brief vector_value Calculates the vector of mobility_cahn_hilliard.
-   * @param field_vectors Vectors of the fields on which the mobility_cahn_hilliard may depend.
-   * @param property_vector Vectors of the mobility_cahn_hilliard values
+   * @brief Calculates the vector of mobility_cahn_hilliard.
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:103]:
```suggestion
   * @brief Calculate the vector of mobility_cahn_hilliard.
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 4]
```diff
@@ -79,9 +88,10 @@ class MobilityCahnHilliardModelConstant : public MobilityCahnHilliardModel
   }
 
   /**
-   * @brief value Computes mobility_cahn_hilliard.
-   * @param fields_value Value of the various field on which the property may depend.
-   * @return value of the physical property calculated with the fields_value
+   * @brief Computes the mobility.
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:91]:
```suggestion
   * @brief Compute the mobility.
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 5]
```diff
@@ -20,57 +20,66 @@
...
   /**
-   * @brief Destructor of derived class
+   * @brief Destructor of derived class.
    */
   ~MobilityCahnHilliardModelConstant() = default;
 
   /**
-   * @brief Method to access the mobility constant, though it returns the same value as the value function, it is implemented here for generality.
-   * @return value of the mobility constant
+   * @brief Returns the mobility constant, though it returns the same
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:79]:
```suggestion
   * @brief Return the mobility constant, though it returns the same
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 6]
```diff
@@ -104,13 +115,15 @@ class MobilityCahnHilliardModelConstant : public MobilityCahnHilliardModel
   }
 
   /**
-   * @brief jacobian Calculates the jacobian (the partial derivative) of the density with respect to a field
-   * @param field_values Value of the various fields on which the property may depend.
-   * @param id Indicator of the field with respect to which the jacobian
+   * @brief Calculates the jacobian (the partial derivative) of the
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:118]:
```suggestion
   * @brief Calculate the Jacobian (the partial derivative) of the
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 7]
```diff
@@ -90,9 +100,10 @@ class MobilityCahnHilliardModelConstant : public MobilityCahnHilliardModel
   }
 
   /**
-   * @brief vector_value Calculates the vector of mobility_cahn_hilliard.
-   * @param field_vectors Vectors of the fields on which the mobility_cahn_hilliard may depend.
-   * @param property_vector Vectors of the mobility_cahn_hilliard values
+   * @brief Calculates the vector of mobility_cahn_hilliard.
+   * @param[in] field_vectors Vectors of the fields on which the mobility
+   * may depend.
+   * @param[out] property_vector Vectors of the mobilityvalues
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:106]:
```suggestion
   * @param[out] property_vector Vectors of the mobility values
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 8]
```diff
@@ -139,13 +154,19 @@ class MobilityCahnHilliardModelConstant : public MobilityCahnHilliardModel
...
+ * where D is the mobility constant.
  */
 class MobilityCahnHilliardModelQuartic : public MobilityCahnHilliardModel
 {
 public:
   /**
-   * @brief Default constructor
+   * @brief Default constructor.
+   *
+   * @param[in] p_mobility_cahn_hilliard_constant the user chosen mobility
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:168]:
```suggestion
   * @param[in] p_mobility_cahn_hilliard_constant the user defined mobility
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 9]
```diff
@@ -155,13 +176,13 @@ class MobilityCahnHilliardModelQuartic : public MobilityCahnHilliardModel
...
   /**
-   * @brief Destructor of derived class
+   * @brief Destructor of derived class.
    */
   ~MobilityCahnHilliardModelQuartic() = default;
 
   /**
-   * @brief Method to access the mobility constant, though it returns the same value as the value function, it is implemented here for generality
-   * @return value of the mobility constant
+   * @brief Returns the mobility constant.
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:184]:
```suggestion
   * @brief Return the mobility constant.
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 10]
```diff
@@ -170,9 +191,10 @@ class MobilityCahnHilliardModelQuartic : public MobilityCahnHilliardModel
   }
 
   /**
-   * @brief value Calculates the mobility_cahn_hilliard.
-   * @param fields_value Value of the various field on which the property may depend.
-   * @return value of the physical property calculated with the fields_value.
+   * @brief Calculates the mobility.
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:194]:
```suggestion
   * @brief Calculate the mobility.
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 11]
```diff
@@ -170,9 +191,10 @@ class MobilityCahnHilliardModelQuartic : public MobilityCahnHilliardModel
...
+   * @param[in] fields_value Value of the various field on which the mobility
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:195]:
```suggestion
   * @param[in] fields_value Value of the various fields on which the mobility
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 12]
```diff
@@ -190,9 +212,10 @@ class MobilityCahnHilliardModelQuartic : public MobilityCahnHilliardModel
   }
 
   /**
-   * @brief vector_value Calculates the vector of mobility_cahn_hilliard_constant.
-   * @param field_vectors Vectors of the fields on which the mobility_cahn_hilliard_constant may depend.
-   * @param property_vector Vectors of the mobility_cahn_hilliard_constant values
+   * @brief Calculates the vector of mobility.
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:215]:
```suggestion
   * @brief Calculate the vector of mobility.
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 13]
```diff
@@ -212,11 +235,14 @@ class MobilityCahnHilliardModelQuartic : public MobilityCahnHilliardModel
   }
 
   /**
-   * @brief jacobian Calculates the jacobian (the partial derivative) of the density with respect to a field
-   * @param field_values Value of the various fields on which the property may depend.
-   * @param id Indicator of the field with respect to which the jacobian
+   * @brief Calculates the jacobian (the partial derivative) of the
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:238]:
```suggestion
   * @brief Calculate the Jacobian (the partial derivative) of the
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 14]
```diff
@@ -212,11 +235,14 @@ class MobilityCahnHilliardModelQuartic : public MobilityCahnHilliardModel
...
+   * mobility with respect to a field.
+   * @param[in] field_values Value of the various fields on which the mobility
+   * may depend.
+   * @param[in] id Indicator of the field with respect to which the jacobian
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:242]:
```suggestion
   * @param[in] id Indicator of the field with respect to which the Jacobian
```

File: include/core/mobility_cahn_hilliard_model.h [hunk 15]
```diff
@@ -235,10 +261,13 @@ class MobilityCahnHilliardModelQuartic : public MobilityCahnHilliardModel
   }
 
   /**
-   * @brief vector_jacobian Calculates the derivative of the density with respect to a field.
-   * @param field_vectors Vector for the values of the fields used to evaluate the property.
-   * @param id Identifier of the field with respect to which a derivative should be calculated.
-   * @param jacobian vector of the value of the derivative of the density with respect to the field id.
+   * @brief Calculates the derivative of the mobility with respect to a field.
```
Comment (blaisb) [inline include/core/mobility_cahn_hilliard_model.h:264]:
```suggestion
   * @brief Calculate the derivative of the mobility with respect to a field.
```

File: include/solvers/cahn_hilliard_filter.h [hunk 16]
```diff
@@ -31,25 +31,26 @@ class CahnHilliardFilterBase
 
   /**
    * @brief Instantiates and returns a pointer to a CahnHilliardFilterBase
```
Comment (blaisb) [inline include/solvers/cahn_hilliard_filter.h:33]: Fix s here

File: include/solvers/cahn_hilliard_filter.h [hunk 17]
```diff
@@ -31,25 +31,26 @@ class CahnHilliardFilterBase
...
    */
   static std::shared_ptr<CahnHilliardFilterBase>
   model_cast(const Parameters::CahnHilliard &cahn_hilliard_parameters);
 
   /**
-   * @brief Calculates the value of the filtered phase fraction
-   * @param unfiltered_phase Value of the phase fraction before applying the
-   * filter
-   * @return Value of the phase fraction after applying the filter
+   * @brief Calculates the value of the filtered phase fraction.
```
Comment (blaisb) [inline include/solvers/cahn_hilliard_filter.h:42]:
```suggestion
   * @brief Calculate the value of the filtered phase fraction.
```

File: include/solvers/cahn_hilliard_filter.h [hunk 18]
```diff
@@ -58,7 +59,7 @@ class CahnHilliardNoFilter : public CahnHilliardFilterBase
   {}
 
   /**
-   * @brief Returns the phase fraction with no modification
+   * @brief Returns the phase fraction with no modification.
```
Comment (blaisb) [inline include/solvers/cahn_hilliard_filter.h:62]:
```suggestion
   * @brief Return the phase fraction with no modification.
```

File: include/solvers/cahn_hilliard_filter.h [hunk 19]
```diff
@@ -79,10 +80,10 @@ class CahnHilliardClipFilter : public CahnHilliardFilterBase
   {}
 
   /**
-   * @brief Calculates the value of the filtered phase fraction
-   * @param unfiltered_phase Value of the phase fraction before applying the
-   * filter
-   * @return Value of the phase fraction after applying the filter
+   * @brief Calculates the value of the filtered phase fraction.
```
Comment (blaisb) [inline include/solvers/cahn_hilliard_filter.h:83]:
```suggestion
   * @brief Calculate the value of the filtered phase fraction.
```

File: include/solvers/cahn_hilliard_filter.h [hunk 20]
```diff
@@ -104,10 +107,10 @@ class CahnHilliardTanhFilter : public CahnHilliardFilterBase
...
```
Comment (blaisb) [inline include/solvers/cahn_hilliard_filter.h:110]:
```suggestion
   * @brief Calculate the value of the filtered phase fraction.
```

**Review by blaisb — CHANGES_REQUESTED** [review]

File: include/core/mobility_cahn_hilliard_model.h (hunk 1 above)
Comment (PierreLaurentinCS) [reply include/core/mobility_cahn_hilliard_model.h:23]: There exist different models to compute the mobility, that is what I mean.

Comment (PierreLaurentinCS) [reply include/core/mobility_cahn_hilliard_model.h:23]: Actually I'll just remove the "general case", it is not required

**Review by blaisb — APPROVED** [review]

**Pull request merged**